src/java/org/apache/cassandra/config/CassandraRelevantProperties.java

```diff
@@ -194,6 +194,11 @@
      */
     UCS_COMPACTION_AGGREGATE_PRIORITIZER("unified_compaction.custom_compaction_aggregate_prioritizer"),
 
+    /**
+     * Makes UCS the default compaction strategy for all tables
+     */
+    UCS_DEFAULT_STRATEGY("cassandra.default.ucs.compaction", "true"),
```

Comment:
do we really want it on by default, rather than only for CNDB?

I mean - if we want it to really be the default, perhaps changing `CompactionParams.DEFAULT` is enough?

Reply:
I agree with this. We could probably use `DatabaseDescriptor.isEmulateDbaasDefaults()` rather than introduce the new flag. This would then only work for CNDB.

WDYT?

Reply:
I very much like the idea.

I don't know for sure if `isEmulateDbaasDefaults` is the right flag, but yeah, we should have a "running in cndb mode" flag somewhere.

Reply:
I'm pretty certain the `isEmulateDbaasDefaults` is the correct option for this. It's what's used by `GuardrailsConfig` to determine whether to use CNDB specific guardrails or on-prem guardrails. It actually makes far more sense to use this flag here because it matches the table options guardrail.